### PR TITLE
add new alert for AAP

### PR DESCRIPTION
**Pull request by songleo**
Add new alerts to monitor AAP:

- AAPMetricMissing: make sure the AAP metrics are valid
- AAPEndpointAddressNotReady: make sure the AAP endpoint is ready

Signed-off-by: Song Song Li <ssli@redhat.com>

### CODE REVIEW

**Review by NemoDang — APPROVED** [review]

**Review by NemoDang — APPROVED** [review]

**Comment by songleo** [comment]
TODO: add the runbook for this new alerts

File: cluster-bootstrap/multicluster-observability/base/custom-alerts/aap-alerts-policy.yaml [hunk 1]
```diff
@@ -40,13 +40,37 @@ spec:
                     groups:
                     - name: AnsibleAutomationPlatform
                       rules:
+                      - alert: AAPMetricMissing
+                        annotations:
+                          description: AAP metric missing, the number of samples should be equal to 44, the current value is {{`{{$value}}`}}
+                        expr: scrape_samples_scraped{job="automation-controller-service",namespace="ansible-automation-platform"} < 44
```
Comment (andreadecorte) [inline cluster-bootstrap/multicluster-observability/base/custom-alerts/aap-alerts-policy.yaml:46]: this means that if AAP expose more metrics in a new version, we should update this number right?

Comment (songleo) [reply cluster-bootstrap/multicluster-observability/base/custom-alerts/aap-alerts-policy.yaml:46]: yes, also need to update our custom metric list.

Comment (songleo) [reply cluster-bootstrap/multicluster-observability/base/custom-alerts/aap-alerts-policy.yaml:46]: if the new version AAP exposes more metrics, we do not update the custom metric list, then we still collect the 44 metrics from AAP.

Comment (andreadecorte) [reply cluster-bootstrap/multicluster-observability/base/custom-alerts/aap-alerts-policy.yaml:46]: but in the custom metrics list we don't have 44 whitelisted metrics, how do you get to 44? whitelisted + base ones?

Comment (songleo) [reply cluster-bootstrap/multicluster-observability/base/custom-alerts/aap-alerts-policy.yaml:46]: I tested in our dev cluster and get 44 AAP metrics.

Comment (songleo) [reply cluster-bootstrap/multicluster-observability/base/custom-alerts/aap-alerts-policy.yaml:46]: I think Prom will add some non-AAP metrics in AAP metric endpoint.

**Comment by andreadecorte** [comment]
lgtm, thanks!

**Review by NemoDang — APPROVED** [review]

**Pull request merged**